settings: Fix filter alignment under headings in all settings panels.

NickeSteenen opened the pull request:
Previously, when resizing the window, the filters/buttons and the header did not fit on the same row, the filters/buttons were pushed onto the next row but not put directly below the heading. 

This changes the settings_panel_list_header and add_default_streams_button_container into flex containers and properly aligns the items below the heading when changing window size. 

Fixes: https://github.com/zulip/zulip/issues/30353 <!-- Issue link, or clear description.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

<details>
 <summary>Screenshots</summary>

Resulting behavior of the filters when resizing the window

<img width="693" alt="Skärmavbild 2024-10-16 kl  18 15 47" src="https://github.com/user-attachments/assets/9fa7259c-39cc-47a8-849e-1a50c89a9aff">

<img width="377" alt="Skärmavbild 2024-10-16 kl  18 15 31" src="https://github.com/user-attachments/assets/b963cd38-6f0f-4d90-936c-5b21361de5e4">

<img width="1033" alt="Skärmavbild 2024-10-16 kl  18 19 09" src="https://github.com/user-attachments/assets/3807bebb-e006-449c-ac57-a1a94e8b01f6">

<img width="890" alt="Skärmavbild 2024-10-16 kl  18 19 29" src="https://github.com/user-attachments/assets/1d089188-544c-4085-bff0-38c9ea500c4e">

</details>


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Thanks!

Hm, it looks to me like there's too much vertical space between "Deactivated users" and the search fields below. How is that space determined?

NickeSteenen commented:
> Thanks!
> 
> Hm, it looks to me like there's too much vertical space between "Deactivated users" and the search fields below. How is that space determined?

Hi, you are right that there is a tiny bit of extra space with a top margin of 12px for this specific panel and the panel called "Default channels" because of the way the height of the dropdown menu and button in these cases are explicitly set. The extra margin when the filters are below the heading is a result of the top margin needed for the filters to be centered properly when floating to the right of the page on the same row as the heading. It might be possible to look into this further and that would probably mean adjusting these specific filters and their sizes.

alya commented:
@sahil839 could you take a look at this one? I haven't tested.

It would be best not to have the extra vertical space, but the change would be a clear improvement even with the extra space.

NickeSteenen commented:
> Current changes look good.
> 
> For removing the extra vertical space, I guess one way can be to remove top and bottom margins for headings, search inputs, buttons in a couple of panels, and just add some bottom margin below the whole flex container. Not sure, but might be worth trying.

I looked into this and managed to get the margins to be consistent everywhere with no extra large gap. I removed the margins for the individual filters and added the margins to the specific container holding them!

sahil839 commented on web/styles/settings.css:
```diff
@@ -725,9 +728,7 @@ input[type="checkbox"] {
         padding-top: 0;
         padding-bottom: 0;
         height: 30px;
-
```
We can avoid unrelated changes like this.

sahil839 commented on web/styles/settings.css:
```diff
@@ -602,8 +602,8 @@ input[type="checkbox"] {
 #add-custom-profile-field-btn {
-    float: right;
     margin-top: 4px;
+    margin-bottom: 4px;
```
Why this has the bottom margin for 4px and 12px? Can we keep the gap same between the button or search input and the top of the table same for all the panels?

NickeSteenen replied:
This is now also 12px as it is for all the other settings panel filters

sahil839 reviewed (commented):
Posted a couple of small comments, but looks good overall. I think having all the changes in a single commit would be better.

One thing that I notice is that for much narrower screen is that there is no gap between the elements if there are two elements on the right side. Not sure how important is fixing the design for such narrow screens, but maybe we can fix that if it is not too complex.
![image](https://github.com/user-attachments/assets/6fc45634-0455-4293-8193-bfd5f775396a)

sahil839 commented on web/styles/settings.css:
```diff
@@ -602,8 +602,9 @@ input[type="checkbox"] {
-    margin-top: 4px;
+    margin-top: auto;
+    margin-bottom: 12px;
+    padding: 4px 14px;
```
We use height property instead of setting in the top and bottom paddings for the "Add default channel" button, but this seems better.

Also, would be good to share the screenshot for this button since its design has been changed.

NickeSteenen replied:
Since I put the "Add default channel" button is inside a flexbox with "align-items" that defaults to "stretch", the height property of that button is not needed right now, and it uses the padding set from the .button class. I can remove the height property from that button.

sahil839 replied:
It works fine for the two font-settings that we support in prod currently, but might break things at larger font-sizes where the search input height is still fixed. But should be fine for now.

NickeSteenen commented:
Here are some screenshots of the before and after of the "Custom profile fields" button 

<details>
 <summary>Screenshots</summary>
Before:

![before](https://github.com/user-attachments/assets/d95ad63e-a391-491f-a999-621dc7b9c302)

After:

![after](https://github.com/user-attachments/assets/a98703b8-88dd-4651-8f57-b5627db07ded)

</details>

sahil839 commented:
@timabbott I think this is ready for you to have a look.
